patches/tModLoader/Terraria/ModLoader/UI/DownloadManager/DownloadFile.cs

```diff
@@ -42,7 +42,7 @@ internal class DownloadFile
 			return true;
 		}
 
-		public Task<DownloadFile> Download(CancellationToken token, Action<float> updateProgressAction = null) {
+		public Task<DownloadFile> Download(CancellationToken token, Action<float, float, float> updateProgressAction = null) {
```

Comment:
Why is this a <float, float, float>, if you're passing float, int and long to it? Just make it <float, long, long>